src/fe-common/core/fe-tls.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2015 Alexander Færøy <ahf@irssi.org>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301,USA
+ */
+
+#include "module.h"
+#include "signals.h"
+#include "settings.h"
+#include "levels.h"
+#include "tls.h"
+
+#include "module-formats.h"
+#include "printtext.h"
+
+#include "fe-tls.h"
+
+static void tls_handshake_finished(SERVER_REC *server, TLS_REC *tls)
+{
+	if (! settings_get_bool("tls_verbose_connect"))
+		return;
+
+	printformat(server, NULL, MSGLEVEL_CLIENTNOTICE, TXT_TLS_CERT_HEADER);
+
+	GSList *certs = NULL;
+	for (certs = tls->certs; certs != NULL; certs = certs->next) {
+		TLS_CERT_REC *tls_cert_rec = certs->data;
+
+		printformat(server, NULL, MSGLEVEL_CLIENTNOTICE, TXT_TLS_CERT_SUBJECT_HEADER);
+
+		GSList *subject = NULL;
+		for (subject = tls_cert_rec->subject; subject != NULL; subject = subject->next) {
+			TLS_CERT_ENTRY_REC *subject_data = subject->data;
+			printformat(server, NULL, MSGLEVEL_CLIENTNOTICE, TXT_TLS_CERT_NAMED_ENTRY, subject_data->name, subject_data->value);
+		}
+
+		printformat(server, NULL, MSGLEVEL_CLIENTNOTICE, TXT_TLS_CERT_ISSUER_HEADER);
+
+		GSList *issuer = NULL;
+		for (issuer = tls_cert_rec->issuer; issuer != NULL; issuer = issuer->next) {
+			TLS_CERT_ENTRY_REC *issuer_data = issuer->data;
+			printformat(server, NULL, MSGLEVEL_CLIENTNOTICE, TXT_TLS_CERT_NAMED_ENTRY, issuer_data->name, issuer_data->value);
+		}
+	}
+
+	printformat(server, NULL, MSGLEVEL_CLIENTNOTICE, TXT_TLS_PROTOCOL_VERSION, tls->protocol_version);
+	printformat(server, NULL, MSGLEVEL_CLIENTNOTICE, TXT_TLS_CIPHER, tls->cipher_size, tls->cipher);
+
+	printformat(server, NULL, MSGLEVEL_CLIENTNOTICE, TXT_TLS_VALIDATION_PERIOD, tls->not_before, tls->not_after);
+
+	printformat(server, NULL, MSGLEVEL_CLIENTNOTICE, TXT_TLS_PUBLIC_KEY, tls->public_key_size, tls->public_key_algorithm);
```

Comment:
Not sure if it's worth doing anything about this, but all these `printformat(server, NULL, MSGLEVEL_CLIENTNOTICE,` are a bit too noisy visually.

Wrapping the lines so that i don't have to scroll horizontally to read the important part would be nice too, i guess.